### DISCUSSION TITLE
JavaScript debugging in Codespaces

### DISCUSSION THREAD

**Discussion by maiertech** [community/community#3536] [Codespaces]
I tested Javascript debugging (with vscode-js-debug) in Codespaces and noticed that it does not work. Here is an example to demonstrate what did not work.

First I debugged my project locally. I set a breakpoint and launched the debugger via [easy NPM script debugging](https://github.com/microsoft/vscode-js-debug#easy-npm-script-debugging). This screenshot shows my breakpoint and it shows that I can hover over variables to see their value:

<img width="734" alt="local" src="https://user-images.githubusercontent.com/1482402/110996067-5d885880-8349-11eb-8fc6-3b72be82c4e9.png">

Then I launched my project in Codespaces using the `javascript-node:14` image, set a breakpoint at the exact same spot and then launched the debugger the exact same way as before. The debugger hits the breakpoint and stops, but when I hover over the same variable, I see a TypeScript definition and not a value:

<img width="743" alt="codespaces" src="https://user-images.githubusercontent.com/1482402/110996409-e30c0880-8349-11eb-96d2-bde9de23f259.png">

This happens both, when I run the Codespace in VS Code or in the browser. It's important to mention that the debugger does not offer stepping through when paused. The controls look like this:

<img width="351" alt="Screen Shot 2021-03-12 at 3 50 43 PM" src="https://user-images.githubusercontent.com/1482402/110997110-eb187800-834a-11eb-98be-fb506440bf32.png">

And it's also not possible to display any values in the debug console.

Could the Codespaces team clarify whether this is supposed to work or whether it's expected that it does not work? And maybe you can add a little technical background of why it can or cannot work.

Thanks!

**Comment by srivatsn** [comment]
Tagging @weinand who might be able to help.

**Reply by Chuxel** [reply]
I actually wouldn't expect to see a difference here, but @connor4312 may be able to confirm one way or another. 

@maiertech can you post your VS Code version information? (Contents from Help > About > Copy)

**Reply by connor4312** [reply]
@isidorn is the expert on hovers 🙂

**Reply by isidorn** [reply]
The hover behaviour should not depend on if you are running in a Codespace or just VS Code.
You should not be able to see the language hover while you are debugging.
Unless you are holding down the `alt` key when moving the mouse over - then the language hover appears.

@maiertech  If you can reproduce this  - that the language hover is shown while you are debugging in VS Code it would be great if you simple file a new issue https://github.com/microsoft/vscode/issues Feel free to ping me @isidorn and I can investigate further. Thanks!
